index.html

```diff
@@ -1879,6 +1879,88 @@ <h3>Versioning Cryptography Suites</h3>
         </p>
       </section>
 
+      <section>
+        <h3>Layer Cryptographic Options Appropriately</h3>
+
+        <p>
+Modern cryptographic algorithms provide a number of tunable parameters and
+options to ensure that the algorithms can meet different use case requirements.
+For example, embedded systems have limited processing and memory environments
+and might not have the resources to generate the strongest digital signatures
+for a given algorithm. Other environments, like financial trading systems,
+might only need to protect data for a day while the trade is occuring while
+other environments might need to protect data for multiple decades. To meet
+these needs, cryptographic algorithm designers often provide multiple ways to
+configure a cryptographic algorithm.
+        </p>
+        <p>
+Cryptographic library implementers often take the specifications created by
+cryptographic algorithm designers and specification authors and implement them
+such that all options are available to the application developers that use their
+libraries. This can be due to not knowing which combination of features a
+particular application developer might need for a given cryptographic algorithm.
+All options are often exposed to application developers.
+        </p>
+        <p>
+Application developers that use cryptographic libraries often do not have the
+same level of knowledge as cryptographic algorithm designers, specification
+authors, or cryptographic library implementers. This lack of knowledge can lead
+to an inappropriate selection of cryptographic parameters and options for a
+particular application.
+        </p>
+        <p>
+This specification sets the priority of constituencies to protect application
+developers over cryptographic library implementers over cryptographic
+specification authors over cryptographic algorithm designers. Given these
+priorities, the following recommendations are made:
+        </p>
+        <ul>
+          <li>
+Cryptographic algorithm designers are advised to minimize the number of
```

Comment:
having this kind of suggestion in the spec really feels like the tail wagging the dog. I would be more comfortable focusing just on implementers of the spec (which are often "cryptographic library implementers")

Reply:
Yeah, maybe this is meant to be "Cryptographic suite designers".

Reply:
This section is intended to convey a full top-to-bottom philosophy. I'll note that this conversation has surfaced in CFRG before and [is currently being debated](https://mailarchive.ietf.org/arch/msg/cfrg/FpfQFAtqwn5gHPV619RLvMAu98I/). I believe that not including the foundational work that the IETF CFRG does would be problematic as there is a feedback cycle that exists between the high-level security protocols and the low-level security primitives. One of the problems with the current state of the industry is in the myriad of options that are provided at the low-level and then are needlessly exposed to application developers that don't know any better.

Explaining the complete philosophy is better than just explaining part of the philosophy, IMHO.

That said, we might be missing a layer, as "cryptographic library implementers" (libcrypto) are different from "cryptographic suite designers" (eddsa-2022), which are different from "cryptographic suite implementers" (di.js).

I'll try to take this into account and rewrite bits of this section. Suggestions at concrete text will help if you guys have any.

Reply:
> not including the foundational work that the IETF CFRG does would be problematic as there is a feedback cycle that exists between the high-level security protocols and the low-level security primitives

Agree here. Is there a middle ground where we can reference (or directly quote) an IETF document to not seem as if we're making these prescriptions ourselves?

Reply:
Fixed in 1a508e62597d326f1ace75c594df29ec53eac56c.